File: Java/Aula17/src/aula17/Caneta.java

package aula17;


public class Caneta {
    public String modelo;
    private float ponta;
    private String cor;
    public boolean tampada;
    
    public Caneta(String m, String c, float p){
        this.tampar();
        this.modelo = m;
        this.cor = c;
        this.setPonta(p);
    }
    
    public String getModelo(){
        return this.modelo;
    }
    public void setModelo(String m){
        this.modelo = m;
    }
    public String getCor(){
        return this.cor;
    }
    public void setCor(String c){
        this.cor = c;
    }
    public float getPonta(){
        return this.ponta;
    }
    public void setPonta(float p){
        this.ponta = p;
    }
    
    public void destampar(){
        this.tampada = false;
    }
    
    public void tampar(){
        this.tampada = true;
    }
    
    public void status(){
        System.out.println("Atributos da Caneta:");
        System.out.println("Modelo: " + this.getModelo());
        System.out.println("Cor: " + this.getCor());
        System.out.println("Ponta: " + this.getPonta());
        System.out.println("Tampada: " + this.tampada + "\n");
    }
    
}

File: PHP/Aula30/Caneta.php
<?php

class Caneta {

    public $modelo;
    public $cor;
    private $ponta;
    protected $carga;
    protected $tampada;
    
    public function getModelo() {
        return $this->modelo;
    }

    public function getPonta() {
        return $this->ponta;
    }

    public function setModelo($modelo) {
        $this->modelo = $modelo;
    }

    public function setPonta($ponta) {
        $this->ponta = $ponta;
    }

    public function __construct($modelo, $cor, $ponta) {
        $this->modelo = $modelo;
        $this->cor = $cor;
        $this->ponta = $ponta;
        $this->tampar();
    }

        public function rabiscar(){
        if ($this->tampada == true) {
            echo "<p>Destampe a Caneta para Rabiscar.";
        } else {
            echo "<p>Rabiscos...</p>";
        }
    }
    
    public function tampar(){
        $this->tampada = true;
    }
    
    public function destampar(){
        $this->tampada = false;
    }
}